test/jdk/java/awt/Container/MouseEnteredTest.java

```diff
@@ -0,0 +1,243 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.EventQueue;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+import javax.swing.ButtonGroup;
+import javax.swing.JCheckBoxMenuItem;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPopupMenu;
+import javax.swing.JRadioButtonMenuItem;
+import javax.swing.JTextArea;
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 4159745
+ * @key headful
+ * @summary Mediumweight popup dragging broken
+ * @run main MouseEnteredTest
+ */
+
+public class MouseEnteredTest extends JFrame implements ActionListener {
+    static volatile MouseEnteredTest test;
+    static volatile Point p;
+    static volatile Point p2;
+
+    static String strMotif = "Motif";
+    static String motifClassName = "com.sun.java.swing.plaf.motif.MotifLookAndFeel";
+    static char cMotif = 'o';
+
+    static String strWindows = "Windows";
+    static String windowsClassName = "com.sun.java.swing.plaf.windows.WindowsLookAndFeel";
+    static char cWindows = 'W';
+
+    static String strMetal = "Metal";
+    static String metalClassName = "javax.swing.plaf.metal.MetalLookAndFeel";
+    static char cMetal = 'M';
+
+    static JMenu m;
+    static JMenu menu;
+
+    static MouseListener ml = new MouseEnteredTest.MouseEventListener();
+
+    public MouseEnteredTest() {
+        JPopupMenu.setDefaultLightWeightPopupEnabled(false);
+        setJMenuBar(getMyMenuBar());
+        getContentPane().add("Center", new JTextArea());
+        setSize(400, 500);
+        setLocationRelativeTo(null);
+        setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                test = new MouseEnteredTest();
+            });
```

Comment:
```suggestion
            EventQueue.invokeAndWait(() -> test = new MouseEnteredTest());
```
Lambda

Reply:
this is just removing the brackets, will leave as is